File: 012_integer_to_roman/Solution.go
package main

import "fmt"

func intToRoman(num int) string {
    values := []int{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1}
    symbols := []string{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"}

    result := ""
    for i, v := range values {
        for num >= 0 && num >= v {
            num -= v
            result += symbols[i]
        }
    }

    return result
}

func main() {
    fmt.Println(intToRoman(140))
}

File: 012_integer_to_roman/Solution.py
class Solution:
    def __init__(self):
        self.values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
        self.symbols = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]

    def intToRoman(self, num: int) -> str:
        result = ''
        for i in range(len(self.values)):
            while num >= 0 and self.values[i] <= num:
                num -= self.values[i]
                result += self.symbols[i]

        return result


if __name__ == '__main__':
    solution = Solution()
    print(solution.intToRoman(140))
